web/src/compose_paste.ts

```diff
@@ -481,9 +482,13 @@ export function paste_handler_converter(
             const code = codeElement.textContent;
             assert(code !== null);
 
-            // We convert single line code inside a code block to inline markdown code,
-            // and the code for this is taken from upstream's `code` rule.
-            if (!code.includes("\n")) {
+            const className = codeElement.getAttribute("class") ?? "";
+            const language = node.parentElement?.classList.contains("zulip-code-block")
+                ? (node.closest<HTMLElement>(".codehilite")?.dataset?.codeLanguage ?? "")
```

Comment:
I recommend a function for this computation; this is hard to read, and I'm not sure all the `?` comprehensions are necessary. Also, avoid `dataset`, as `data-code-language` is more greppable than `codeLanguage`.

Reply:
(But I guess this code isn't new).